design-toolkit/components/src/components/options/index.tsx

```diff
@@ -12,86 +12,185 @@
-        {...props}
-      >
+    <OptionsContext.Provider value={{ size, color }}>
+      <AriaListBox<T> ref={ref} className={list()} {...rest}>
         {children}
       </AriaListBox>
     </OptionsContext.Provider>
   );
 }
 Options.displayName = 'Options';
 
-Options.Item = OptionsItem;
-
-interface OptionsSectionProps<T extends IOptionsItem>
-  extends AriaListBoxSectionProps<T> {
-  header?: string;
-}
-
 export function OptionsSection<T extends IOptionsItem>({
   children,
   header,
   items,
 }: OptionsSectionProps<T>) {
   return (
-    <AriaListBoxSection
-      id={header}
-      className='mt-s border-default-dark border-t first:border-none'
-    >
-      <AriaHeader className='m-xs my-s text-default-dark text-header-xs'>
-        {header}
-      </AriaHeader>
+    <AriaListBoxSection id={header} className={section()}>
+      <AriaHeader className={sectionHeader()}>{header}</AriaHeader>
       <AriaListBoxCollection items={items}>{children}</AriaListBoxCollection>
     </AriaListBoxSection>
   );
 }
 OptionsSection.displayName = 'Options.Section';
+
+export function OptionsItem({
+  children,
+  className,
+  size: sizeProp,
+  color: colorProp,
+  ...props
+}: OptionsItemProps) {
+  const context = useContext(OptionsContext) ?? {};
+
+  const { size, color } = useMemo(
+    () =>
+      mergeProps(context, {
+        size: sizeProp,
+        color: colorProp,
+      }),
+    [context, sizeProp, colorProp],
+  );
+
+  return (
+    <AriaListBoxItem
+      {...props}
+      className={composeRenderProps(className, () => item({ size, color }))}
+    >
+      {composeRenderProps(children, (children) => (
+        <>
+          {typeof children === 'string' ? (
+            <OptionsItemLabel>{children}</OptionsItemLabel>
+          ) : (
+            children
+          )}
+        </>
+      ))}
+    </AriaListBoxItem>
+  );
+}
+OptionsItem.displayName = 'Options.Item';
+
+function OptionsItemContent({
+  children,
+  className,
+  ...props
+}: OptionsItemTextProps) {
+  return (
+    <div {...props} className={itemContent()}>
+      {children}
+    </div>
+  );
+}
+OptionsItemContent.displayName = 'Options.Item.Content';
+
+function OptionsItemLabel({
+  children,
+  className,
+  ...props
+}: OptionsItemTextProps) {
+  return (
+    <AriaText {...props} className='truncate' slot='label'>
+      {children}
+    </AriaText>
+  );
+}
+OptionsItemLabel.displayName = 'Options.Item.Label';
+
+function OptionsItemDescription({
+  children,
+  className,
+  ...props
+}: OptionsItemTextProps) {
+  return (
+    <AriaText
+      {...props}
+      className='truncate'
+      data-slot='description'
+      slot='description'
+    >
+      {children}
+    </AriaText>
+  );
+}
+OptionsItemDescription.displayName = 'Options.Item.Description';
+
+function OptionsItemIcon({
+  children,
+  className,
+  ...props
+}: OptionsItemIconProps) {
+  return (
+    <span className={itemIcon()}>
```

Comment:
I think you can do the same thing we talked about eliminating this component. But if you can't not sure why it would need a wrapper span